### DISCUSSION TITLE
Hi,Is there such a bug in ViewSelector class?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#893] [Ideas] [answered]
Hi, when frontstageprovider uses imodel viewportcontrol to preview the model, I use viewselector component or ImodelConnectedViewselector to switch different viewstates to preview different views.

Viewselector works well when I only use the same imodelconnection.

However, when I switch to a different imodelconnection, I find that the viewstate content stored in ViewSelector always records the viewstate data of the imodelconnection opened for the first time. As a result, clicking on the content in viewselector will cause the program to crash.

Now, I get the source code of viewselector, put it in the user code, and add the following content to realize customization. Then, the customized viewselector works well in an imodel connection or switching different imodelconnections.
（I've only added the code outlined in red）
![图片](https://user-images.githubusercontent.com/26344907/109884220-970cf400-7cb7-11eb-9e53-a53207685c5b.png)

Now I want to know whether I use viewselector in the wrong way or whether there is such a small bug. Thank you.

**Comment by bsteinbk** [comment, marked as answer]
@shenshengyi This is indeed a bug in the component. Thanks for pointing it out, and thanks for the fix!

**Reply by bsteinbk** [reply]
PR https://github.com/imodeljs/imodeljs/pull/894 has been opened to apply the suggested fix.
